SingularityService/src/main/java/com/hubspot/singularity/scheduler/SingularityUsagePoller.java

```diff
@@ -160,6 +159,13 @@ private void shuffleTasksOnOverloadedHosts(Map<SingularitySlaveUsage, List<TaskI
         .sorted((usage1, usage2) -> {
           OverusedResource mostOverusedResource1 = getMostOverusedResource(usage1, getSystemLoadForShuffle(usage1), usage1.getMemoryBytesUsed());
           OverusedResource mostOverusedResource2 = getMostOverusedResource(usage2, getSystemLoadForShuffle(usage2), usage2.getMemoryBytesUsed());
+          if (mostOverusedResource1.resourceType != mostOverusedResource2.resourceType) {
+            if (mostOverusedResource1.resourceType == Type.MEMORY) {
+              return -1;
+            } else {
+              return 1;
+            }
+          }
```

Comment:
Hmm, this short circuits _all_ resource comparisons when the two resources being compared aren't the same. So like, if we're comparing `DISK` vs `CPU`, whichever one is `mostOverusedResource1` would always sort as higher. Is that the intent? Should we scope this logic to the specific scenario when one of the resources being compared is `MEMORY`?

Reply:
We don't have shuffles for disk so far. Type can only be MEM or CPU

Reply:
Ahhh, derp. cool